Source/cg_ControlGrisAudioProcessorEditor.cpp

```diff
@@ -487,6 +487,153 @@ void ControlGrisAudioProcessorEditor::sourcesPlacementChangedCallback(SourcePlac
+{
+    //this first part is the constructor from PolarVector
+    auto const length = std::sqrt(x * x + y * y + z * z);
+    if ((x == 0.0f && y == 0.0f) || length == 0.0f)
+        return {};
+
+    auto elevation = HALF_PI - Radians{ std::acos(std::clamp(z / length, -1.0f, 1.0f)) };
+    auto azimuth = Radians{std::acos(std::clamp(x / std::sqrt(x * x + y * y), -1.0f, 1.0f))* (y < 0.0f ? -1.0f : 1.0f)};
+
+    //then translate by pi/2
+    azimuth = HALF_PI - azimuth;
+    elevation = HALF_PI - elevation;
+
+    //and at this point we have the azimuth and elevation sent to SpatGRIS
+    return { azimuth.get(), elevation.get() };
+}
+
+juce::Point<float> getXyFromDomeAzimuthAndElevation(float azimuth, float elevation)
+{
+    // some of this logic is from Source::computeXY()
+    auto const radius{ elevation / MAX_ELEVATION.get()};
+    auto const position = Source::getPositionFromAngle(Radians{ azimuth }, radius);
+
+    // these other manipulations are from ControlGrisAudioProcessor::parameterChanged() and
+    // Source::computeAzimuthElevation()
+    return { (position.x + 1) / 2, 1 - ((position.y + 1) / 2) };
+}
+
+class GetDomeAzimuthAndElevationFromPositionTest : public juce::UnitTest
+{
+public:
+    GetDomeAzimuthAndElevationFromPositionTest() : juce::UnitTest("GetAzimuthAndElevationFromPosition Test") {}
+
+    void runTest() override
+    {
+        beginTest("Test with (0, 0.640747, 0.767752)");
+        {
+            auto const [azim, elev] = getAzimuthAndElevationFromDomeXyz(0.f, 0.640747f, 0.767752f);
+            expectWithinAbsoluteError(azim, 0.f, 0.001f);
+            expectWithinAbsoluteError(elev, 0.69547f, 0.001f);
+
+            auto const cartesianPosition = getXyFromDomeAzimuthAndElevation(azim, elev);
+            expectWithinAbsoluteError(cartesianPosition.x, .5f, 0.001f);
+            expectWithinAbsoluteError(cartesianPosition.y, 0.721375f, 0.001f);
+        }
+    }
+};
+
+static GetDomeAzimuthAndElevationFromPositionTest getAzimuthAndElevationFromPositionTest;
+
+//==============================================================================
+void ControlGrisAudioProcessorEditor::speakerSetupSelectedCallback(const juce::File& speakerSetupFile)
+{
+    auto const showError = [](juce::String error) {
+        juce::AlertWindow::showMessageBox(juce::AlertWindow::WarningIcon,
+                                          "Error Loading Speaker Setup File",
+                                          error,
+                                          "OK");
+    };
+
+    // make sure our file exists
+    if (!speakerSetupFile.existsAsFile()) {
+        showError("This file does not exist: " + speakerSetupFile.getFullPathName());
+        return;
+    }
+
+    // and is a valid speaker setup file
+    auto const speakerSetup = juce::ValueTree::fromXml(speakerSetupFile.loadFileAsString());
+    if (!speakerSetup.isValid() || speakerSetup.getType().toString() != "SPEAKER_SETUP" || speakerSetup.getNumChildren() < 1) {
```

Comment:
since there's already a lot of string constants e.g. `extern juce::String const FIXED_POSITION_DATA_TAG;` etc. do we want to standardize this and use these constants everywhere instead of raw strings? 
It may make it simpler for someone who wants to understand the save file format as this way there's only one place to look for strings instead of grepping through the entire codebase.